src/main/java/de/tum/in/www1/artemis/security/JiraAuthenticationProvider.java

```diff
@@ -134,14 +145,14 @@ else if (e.getStatusCode().is5xxServerError()) {
             }
         }
 
-        if (authenticationResponse != null) {
-            Map content = authenticationResponse.getBody();
-            final String login = (String) content.get("name");
-            final String emailAddress = (String) content.get("emailAddress");
-            User user = userRepository.findOneByLogin(login).orElseGet(() -> userService.createUser(login, "", (String) content.get("displayName"), "", emailAddress, null, "en"));
-            user.setEmail(emailAddress);
+        if (authenticationResponse != null && authenticationResponse.getBody() != null) {
+            final var content = authenticationResponse.getBody();
+            final var login = content.getName();
+            final var emailAddress = content.getEmailAddress();
+            final var user = userRepository.findOneByLogin(login).orElseGet(() -> userService.createUser(login, password, content.getDisplayName(), "", emailAddress, null, "en"));
```

Comment:
```suggestion
            final var user = userRepository.findOneByLogin(login).orElseGet(() -> userService.createUser(login, "", content.getDisplayName(), "", emailAddress, null, "en"));
```

Reply:
we don't want to store the password of JIRA users in our db!!!

Reply:
reverted